### ISSUE TITLE
PEP8 compliance

### ISSUE THREAD

**Issue by craneworks** [fail2ban/fail2ban#876]
Today is so much action I thought I could also open a new ticket. ;)

I would like to see a stop in developing new features and clean up the code to be in compliance with PEP8. 

Why should this be done?
ATM most people are having their own coding styles which results in funny stuff like `var1+<tab><tab><tab>var2<space>=var3`. This is just one example but I found really weird looking stuff.
So the benefits would be:
- the code is getting cleaner
- easier to submit patches from developers
- some strange code may get killed
- the code is getting a full review

On the opposite...
- this will result in a lot of work. A modern IDE (even vim can handle stuff like this with the right plugins) will reduce the amount of work. I fixed most of the PEP8 errors in just a few days.
- features, patches and so on have to be rebased before we can add them. But this is only a one time job.

I think the best way would  be starting by killing each PEP8 Error one by one instead of fixing it file by file.

**Comment by leeclemens** [comment]
I agree, overall, but the issue isn't that IDE's are smarter (even vim) but that github doesn't use blame to ignore whitespace. This creates confusion when looking at old commits where functionality was changed but recent whitespace-only updates were made.

**Comment by rumpelsepp** [comment]
Just append `?w=1` to the URLs and everything is fine:
https://github.com/blog/967-github-secrets

So IMO that's not really a reason.

**Comment by leeclemens** [comment]
@rumpelsepp thanks for that info!

**Comment by leeclemens** [comment]
My previous PEP8 related PRs were met with sarcastic remarks from other fail2ban developers.

**Comment by sebres** [comment]
I have too many large branches, so toooo many conflicts on active development.
In which case the effort (and my cost) does not fit to the profit.
I think, the time was simply out of line.
Sorry if I was sometimes too hard.

**Comment by yarikoptic** [comment]
The verdict was/is/will be:   changes (either new code, or lines touched in changes) should generally aim to comply with PEP8. Exclusions are:

- we are using tabs instead of spaces for indentation
- code lines are allowed to slightly exceed a norm of 79 BUT should caution you about that there might be a need in refactoring.

For everyone I would advise to watch/listen to https://www.youtube.com/watch?v=wf-BqAjZb8M from the father of PEP8.  With that I will close this issue since it can't otherwise be "fixed" at once.
